hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/MapCodecTemplate.java

```diff
@@ -843,4 +843,20 @@ Object addEntryListenerWithPredicate(String name, Data predicate, boolean includ
     @Since("1.4")
     Object addNearCacheInvalidationListener(String name, int listenerFlags, boolean localOnly);
 
+
+    /**
+     * Fetches the specified number of entries from the specified partition starting from specified table index
+     * that match the predicate and applies the projection logic on them.
+     *
+     * @param name       Name of the map
+     * @param tableIndex The slot number (or index) to start the iterator
+     * @param batch      The number of items to be batched
+     * @param projection projection to transform the entries with
+     * @param predicate  predicate to filter the entries with
+     * @return last index processed and list of entries after applied to the projection
+     */
+    @Request(id = 70, retryable = true, response = ResponseMessageConst.DATA, partitionIdentifier = "partitionId")
```

Comment:
I could but I thought this was a general preference for returning various return types, so the `ResponseTemplate` doesn't grow uncontrollably with various different types for different operations. And I saw that some tasks use this serialize->return Data->deserialize technique. 
Or is adding a new `ResponseTemplate` better?